lib/socket_drano.ex

```diff
@@ -284,6 +284,27 @@ defmodule SocketDrano do
     end)
   end
 
+  # drain only `percentage` of the sockets from the socket, and drain them by
+  # percentage `batch_size` every `time`ms
+  defp drain_sockets({:drain_only, percentage, batch_size_percent, time}, count) do
+    socket_drain_size = ceil(count * percentage / 100)
+    socket_drain_batch_size = ceil(socket_drain_size * batch_size_percent / 100)
+
+    sockets = :ets.tab2list(@table)
+
+    Logger.info(
+      "Draining #{percentage}% of sockets total:#{socket_drain_size} in batches of #{socket_drain_batch_size} every #{time}ms"
+    )
+
+    sockets
+    |> Enum.take(socket_drain_size)
+    |> Enum.chunk_every(socket_drain_batch_size)
+    |> Enum.with_index()
+    |> Enum.each(fn {batch, idx} ->
+      spawn(fn -> drain_socket_batch(batch, idx, idx * time) end)
```

Comment:
I notice that the rest of the code uses `spawn`, so that is probably appropriate for this project if the goal is to get it merged upstream someday (if there is even an active upstream?). But a lesson our team has learned over the past couple of years is that supervised processes are a lot easier to manage at scale. Just a comment.

Reply:
Yeah, that was my idea, to keep it the same way, but yeah, no answer from upstream, so I will update it to use tasks and supervisors to not bloat the server with too many processes